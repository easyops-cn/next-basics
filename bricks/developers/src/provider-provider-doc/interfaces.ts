import {
  Type,
  ParameterReflection,
  Reflection,
} from "typedoc/dist/lib/serialization/schema";
import { I18nData } from "@next-core/brick-types";

interface BaseFields {
  headers: Record<string, string>;
  body?: string;
}

interface RequestFields extends BaseFields {
  uri?: string;
  method?: string;
}

export interface ExampleItem {
  description?: I18nData;
  request: RequestFields;
  response: BaseFields;
}
export interface ProcessedProviderDoc {
  serviceId: string;
  brickName: string;
  comment: string;
  endpoint: string;
  parameters: ParameterReflection[];
  returns: Type;
  references: Map<string | number, Reflection>;
  usedReferenceIds: number[];
  examples: ExampleItem[];
}
