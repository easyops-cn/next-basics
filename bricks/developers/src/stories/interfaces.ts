import {
  Story as LegacyStory,
  Chapter as StoryChapter,
  I18nString,
} from "@next-core/brick-types";

// 由于在stories中有很多构建story引用了下面的类型，所以先暂不处理

export interface Chapter extends StoryChapter {}

export interface Story extends LegacyStory {
  category?: string;
}
