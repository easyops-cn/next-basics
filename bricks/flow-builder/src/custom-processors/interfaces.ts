export interface FlowConfig {
  id: string;
  action: Record<string, any>;
  parents?: string[];
  condition?: string;
  children?: FlowConfig[];
  type?: string; // nodeType used by graph
  flowType?: string;
  [key: string]: any;
}

export interface Edge {
  source: string;
  target: string;
  type: string;
  [key: string]: any;
}

export interface CategoryNode {
  key: string;
  title: string;
  category: "category" | "item";
  children?: CategoryNode[];
}

export interface Example {
  description: {
    zh: string;
    en: string;
  };
  request: any;
  response: any;
}

export interface ExtField {
  name?: string;
  source?: "body" | "query";
}

export interface ContractData {
  examples: Example[];
  endpoint: {
    ext_fields?: ExtField[];
    method: string;
    uri: string;
  };
}

export interface Field {
  name: string;
  type?: string;
  description?: string;
  ref?: string;
  fields?: Field[];
  // 前端计算的 fields
  __fields__?: Field[];
}

export interface ModelDefinition {
  name: string;
  fields?: Field[];
}
