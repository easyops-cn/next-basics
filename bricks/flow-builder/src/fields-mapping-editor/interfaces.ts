export interface FieldItem {
  name: string;
  key?: string;
  type: string;
  source?: string;
  description?: string;
  required?: boolean;
  value?: unknown;
  fields?: FieldItem[];
  disabledEdit?: boolean;
}

export interface SimplifiedFieldItem {
  name: string;
  type: string;
  value?: unknown;
  source?: string;
  fields?: SimplifiedFieldItem[];
}

export type FlatFieldChildrenMap = Record<string, FieldItem[]>;
