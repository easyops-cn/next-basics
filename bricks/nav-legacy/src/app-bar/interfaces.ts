import { DesktopItemDir } from "@next-core/brick-types";

export type NormalizedDesktopDir = Pick<DesktopItemDir, "name" | "items"> & {
  size?: string;
};
