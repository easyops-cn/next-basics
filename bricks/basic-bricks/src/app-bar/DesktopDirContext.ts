import React from "react";
import { NormalizedDesktopDir } from "./interfaces";

export interface DirWithCoordinates {
  dir: NormalizedDesktopDir;
  coordinates: {
    x: number;
    y: number;
  };
  activeIndex: number;
}

export interface ContextOfDesktopDir {
  desktopDir?: DirWithCoordinates;
  setDesktopDir?: React.Dispatch<DirWithCoordinates>;
}

export const DesktopDirContext = React.createContext<ContextOfDesktopDir>({});

export const useDesktopDirContext = (): ContextOfDesktopDir =>
  React.useContext(DesktopDirContext);
