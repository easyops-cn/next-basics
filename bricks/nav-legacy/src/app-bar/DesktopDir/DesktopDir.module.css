.dirLink {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(2, 1fr);
  width: 80px;
  height: 80px;
  color: #fff;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.15);
  border-radius: 10px;
  font-size: 21px;
  padding: 7px;
  /* overflow: hidden; */

  &.small {
    width: 56px;
    height: 56px;
  }

  .appIcon,
  :global(.anticon) {
    width: 33px;
    height: 33px;
  }

  :global(.anticon) {
    display: flex;
    justify-content: center;
    align-items: center;
    overflow: hidden;
  }

  &:hover {
    color: #fff;
  }
}

.dirName {
  display: -webkit-box;
  font-size: 16px;
  line-height: 1.5;
  height: 48px;
  text-align: center;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;
  overflow: hidden;
  text-overflow: ellipsis;
}

@media (max-height: 860px), (max-width: 1280px) {
  .dirLink {
    transform: scale(0.75);
  }
  .dirName {
    font-size: 14px;
    height: 42px;
  }
}

@media (max-height: 700px), (max-width: 1024px) {
  .dirLink {
    transform: scale(0.6);
  }
  .dirName {
    font-size: 12px;
    height: 36px;
  }
}

@media (max-height: 640px), (max-width: 900px) {
  .dirLink {
    transform: scale(0.5);
  }
}
