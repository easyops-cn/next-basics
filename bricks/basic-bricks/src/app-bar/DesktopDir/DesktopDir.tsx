import React from "react";
import classNames from "classnames";
import { useDesktopDirContext } from "../DesktopDirContext";
import styles from "./DesktopDir.module.css";
import defaultAppIcon from "../../images/default-app-icon.png";
import { NormalizedDesktopDir } from "../interfaces";

export function DesktopDir(props: NormalizedDesktopDir): React.ReactElement {
  const { setDesktopDir } = useDesktopDirContext();
  const handleDirClick = (e: React.MouseEvent): void => {
    e.stopPropagation();
    // 记录鼠标点击的位置，在文件夹展开时以此为原点做动画。
    setDesktopDir({
      dir: {
        name: props.name,
        items: props.items,
      },
      coordinates: {
        x: e.clientX,
        y: e.clientY,
      },
      activeIndex: -1,
    });
  };

  return (
    <>
      <a
        className={classNames(styles.dirLink, {
          [styles[props.size]]: !!props.size,
        })}
        onClick={handleDirClick}
        role="button"
      >
        {props.items
          .slice(0, 4)
          .map((item) =>
            item.type === "app" ? (
              <img
                className={styles.appIcon}
                key={item.id}
                src={
                  item.app.icons && item.app.icons.large
                    ? /^(?:https?|data):|^\//.test(item.app.icons.large)
                      ? item.app.icons.large
                      : `${window.PUBLIC_ROOT || ""}micro-apps/${item.id}/${
                          item.app.icons.large
                        }`
                    : defaultAppIcon
                }
              />
            ) : (
              <img
                className={styles.appIcon}
                key={item.id}
                src={defaultAppIcon}
              />
            )
          )}
      </a>
      <span className={styles.dirName}>{props.name}</span>
    </>
  );
}
