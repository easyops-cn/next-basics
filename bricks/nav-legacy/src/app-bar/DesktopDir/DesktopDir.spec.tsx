import React from "react";
import { shallow } from "enzyme";
import { DesktopDir } from "./DesktopDir";
import { NormalizedDesktopDir } from "../interfaces";
import * as context from "../DesktopDirContext";

const spyOnSetDesktopDir = jest.fn();
jest.spyOn(context, "useDesktopDirContext").mockReturnValue({
  setDesktopDir: spyOnSetDesktopDir,
});

describe("DesktopDir", () => {
  it("should work", () => {
    const dir: NormalizedDesktopDir = {
      name: "hello",
      items: [
        {
          type: "app",
          id: "hello",
          app: {
            id: "hello",
            name: "world",
          } as any,
        },
      ],
    };
    const wrapper = shallow(<DesktopDir name={dir.name} items={dir.items} />);
    expect(wrapper).toMatchSnapshot();

    const stopPropagation = jest.fn();
    wrapper.find("a").invoke("onClick")({
      stopPropagation,
      clientX: 1,
      clientY: 2,
    } as any);
    expect(stopPropagation).toBeCalled();
    expect(spyOnSetDesktopDir).toBeCalledWith({
      dir,
      coordinates: {
        x: 1,
        y: 2,
      },
      activeIndex: -1,
    });
  });

  it("should render relative url icon", () => {
    const dir: NormalizedDesktopDir = {
      name: "hello",
      items: [
        {
          type: "app",
          id: "hello",
          app: {
            id: "hello",
            name: "world",
            icons: {
              large: "large.png",
            },
          } as any,
        },
      ],
    };
    const wrapper = shallow(<DesktopDir name={dir.name} items={dir.items} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render absolute url icon", () => {
    const dir: NormalizedDesktopDir = {
      name: "hello",
      items: [
        {
          type: "app",
          id: "hello",
          app: {
            id: "hello",
            name: "world",
            icons: {
              large: "/large.png",
            },
          } as any,
        },
      ],
    };
    const wrapper = shallow(<DesktopDir name={dir.name} items={dir.items} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a custom item", () => {
    const dir: NormalizedDesktopDir = {
      name: "hello",
      items: [
        {
          type: "custom",
          id: "hello",
          name: "world",
          url: "/hello",
        },
      ],
    };
    const wrapper = shallow(<DesktopDir name={dir.name} items={dir.items} />);
    expect(wrapper).toMatchSnapshot();
  });
});
