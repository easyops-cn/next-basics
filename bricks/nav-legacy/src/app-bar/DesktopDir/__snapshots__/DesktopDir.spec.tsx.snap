// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopDir should render a custom item 1`] = `
<Fragment>
  <a
    className="dirLink"
    onClick={[Function]}
    role="button"
  >
    <img
      className="appIcon"
      key="hello"
      src="png-url"
    />
  </a>
  <div
    className="dirName"
    title="hello"
  >
    hello
  </div>
</Fragment>
`;

exports[`DesktopDir should render absolute url icon 1`] = `
<Fragment>
  <a
    className="dirLink"
    onClick={[Function]}
    role="button"
  >
    <img
      className="appIcon"
      key="hello"
      src="/large.png"
    />
  </a>
  <div
    className="dirName"
    title="hello"
  >
    hello
  </div>
</Fragment>
`;

exports[`DesktopDir should render relative url icon 1`] = `
<Fragment>
  <a
    className="dirLink"
    onClick={[Function]}
    role="button"
  >
    <img
      className="appIcon"
      key="hello"
      src="micro-apps/hello/large.png"
    />
  </a>
  <div
    className="dirName"
    title="hello"
  >
    hello
  </div>
</Fragment>
`;

exports[`DesktopDir should work 1`] = `
<Fragment>
  <a
    className="dirLink"
    onClick={[Function]}
    role="button"
  >
    <img
      className="appIcon"
      key="hello"
      src="png-url"
    />
  </a>
  <div
    className="dirName"
    title="hello"
  >
    hello
  </div>
</Fragment>
`;
